File: expo-maps/android/src/main/java/expo/modules/maps/records/PointOfInterestRecord.kt
package expo.modules.maps.records

import com.google.android.gms.maps.model.PointOfInterest
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class PointOfInterestRecord(pointOfInterest: PointOfInterest) : Record {
  @Field
  var latLng: LatLngRecord = LatLngRecord(pointOfInterest.latLng)

  @Field
  var name: String = pointOfInterest.name

  @Field
  var placeId: String = pointOfInterest.placeId
}

File: expo-maps/ios/Records/PointOfInterestRecord.swift
import ExpoModulesCore

struct PointOfInterestRecord : Record {
    init() {}

    @Field
    var latLng: [String : Any?]?

    @Field
    var name: String?

    @Field
    var placeId: String?

    init(placeId: String, name: String, location: CLLocationCoordinate2D){
        latLng = LatLngRecord(coordinate: location).toDictionary()
        self.placeId = placeId
        self.name = name
    }
}